graphql_compiler/query_formatting/common.py

```diff
@@ -268,15 +275,15 @@ def ensure_arguments_are_provided(
 
 
 def validate_arguments(
-    expected_types: Mapping[str, GraphQLType], arguments: Mapping[str, Any]
+    expected_types: Mapping[str, GraphQLQueryArgumentType], arguments: Mapping[str, Any]
 ) -> None:
     """Ensure that all arguments are provided and that they are of the expected type."""
     ensure_arguments_are_provided(expected_types, arguments)
     for name in expected_types:
         validate_argument_type(name, expected_types[name], arguments[name])
 
 
-def insert_arguments_into_query(compilation_result, arguments):
+def insert_arguments_into_query(compilation_result: CompilationResult, arguments: Dict[str, Any]):
```

Comment:
I would use the more general `Map` type here, but the documentation of this function and of the inner helper functions clearly specify dict.